### PR TITLE
chore: add release please workflow

### PR DESCRIPTION
**Pull request by pgautier404**

### CODE REVIEW

File: src/Cache/Interceptors/AgentInterceptor.php [hunk 1]
```diff
@@ -10,10 +10,12 @@ class AgentInterceptor extends Interceptor
     private bool $isFirstRequest = true;
     private string $agent;
     private string $runtimeVersion;
+    private string $sdkVersion = "1.10.0"; // x-release-please-version
```
Comment (rishtigupta) [inline src/Cache/Interceptors/AgentInterceptor.php:13]: Do we would need to manually update the sdkVersion whenever we bump the release-please version?

Comment (nand4011) [reply src/Cache/Interceptors/AgentInterceptor.php:13]: I believe release-please knows to update that line based on the `// x-release-please-version`

Comment (rishtigupta) [reply src/Cache/Interceptors/AgentInterceptor.php:13]: Interesting, thanks for letting me know!

**Review by rishtigupta — APPROVED** [review]

File: .github/workflows/on-push-to-main.yaml [hunk 2]
```diff
@@ -49,3 +49,19 @@ jobs:
...
+          token: ${{ secrets.MOMENTO_MACHINE_USER_GITHUB_TOKEN }}
+          release-type: php
+          package-name: momentohq/client-sdk-php
+          extra-files: |
+            src/Cache/Interceptors/AgentInterceptor.php
+          changelog-types: '[{"type":"feat","section":"Features","hidden":false},{"type":"fix","section":"Bug Fixes","hidden":false},{"type":"chore","section":"Miscellaneous","hidden":false}]'
+
+      - name: Publish
+        run: |
+          echo "Published version: ${{ steps.release.outputs.tag_name }}"
```
Comment (nand4011) [inline .github/workflows/on-push-to-main.yaml:66]: Is this to prove it works before adding the actual publishing?

**Review by nand4011 — APPROVED** [review]

File: src/Cache/Interceptors/AgentInterceptor.php (hunk 1 above)
Comment (pgautier404) [reply src/Cache/Interceptors/AgentInterceptor.php:13]: Yep, the `extra-files` option in the gh action tells rp to find and update a version string in this file.

File: .github/workflows/on-push-to-main.yaml (hunk 2 above)
Comment (pgautier404) [reply .github/workflows/on-push-to-main.yaml:66]: Publishing a PHP package that's already indexed in packagist is handled by a github webhook that runs on push. When a new version tag is added, it's automatically added to the index.

**Pull request merged**